pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -189,6 +189,29 @@ func SetDriverCacheMode(disk *Disk) error {
 	return nil
 }
 
+func SetFileDriverType(disk *Disk, diskInfoGetter func(string) (*containerdisk.DiskInfo, error)) error {
+	diskFileFormatUnset := disk.Type == "file" && disk.Source.File != "" && disk.Driver.Type == ""
+	if diskFileFormatUnset {
+		diskInfo, err := diskInfoGetter(disk.Source.File)
+		if err != nil {
+			return err
+		}
+		disk.Driver.Type = diskInfo.Format
+	}
+	// Check also for a backing file
+	backingStore := disk.BackingStore
```

Comment:
qcow2 is great, but it has this history of allowing DOS or security relevant attack angles if not properly sanitized. For containerDisk we explicitly disallow backing files and we ensure that we already do a validation-check on the qcow2 in a confined environment before it ever reaches virt-launcher. Have a look at https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/container-disk/mount.go#L258 and https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/isolation/validation.go#L18.

I would recommend that we do the same for any disk type which saves qcow2. If backing-files are needed, let's maybe analyze after this PR what we can do there, but let's start without them.

Reply:
On the backing files, this only applies for ephemeral disks backed with file-system PVC. These are created by kubevirt and are always raw images. There should be no issue. But I will see a way around to avoid it.

ACK on validation, I will see a way to validate the image on the virt-handler context as well.